shared/test/dataDriver_test.js

```diff
@@ -0,0 +1,222 @@
+      expect(() => new loop.DataDriver({})).to.Throw(/dispatcher/);
+    });
+  });
+
+  describe("#sendTextChatMessage", () => {
+    it("should send a message by updating a chat record", () => {
+      driver.sendTextChatMessage({
+        contentType: CHAT_CONTENT_TYPES.TEXT,
+        message: "Are you there?"
+      });
+
+      let { method, requestBody, url } = requests[0];
+      expect(method).eql("PUT");
+      expect(requestBody).eql('{"timestamp":{".sv":"timestamp"},"value":{"contentType":"chat-text","message":"Are you there?"}}');
+      expect(url.match(/([^\/]+).{12}\.json$/)[1]).eql("chat!00000000");
+    });
+  });
+
+  describe("#fetchServerData", () => {
+    it("should connect to the room on server data", () => {
+      driver.fetchServerData(new actions.FetchServerData({
+        token: "fakeFetch",
+        windowType: "room"
+      }));
+
+      expect(requests[0].url.match(/[^\/]+\/[^\/]+$/)[0]).eql('fakeFetch/.json?limitToLast=1&orderBy="timestamp"');
+      expect(requests[1].url.match(/[^\/]+\/[^\/]+$/)[0]).eql("fakeFetch/meta!lastConnect.json");
+    });
+  });
+
+  describe("#setupWindowData", () => {
+    it("should connect to the room on window data", () => {
+      driver.setupWindowData(new actions.SetupWindowData({
+        roomToken: "fakeSetup"
+      }));
+
+      expect(requests[0].url.match(/[^\/]+\/[^\/]+$/)[0]).eql('fakeSetup/.json?limitToLast=1&orderBy="timestamp"');
+      expect(requests[1].url.match(/[^\/]+\/[^\/]+$/)[0]).eql("fakeSetup/meta!lastConnect.json");
+    });
+  });
+
+  describe("#handleEvent.open", () => {
+    it("should dispatch `DataChannelsAvailable` when the connection is opened", () => {
+      driver.handleEvent({ type: "open" });
+
+      sinon.assert.called(dispatcher.dispatch);
+      sinon.assert.calledWithExactly(dispatcher.dispatch,
+        new actions.DataChannelsAvailable({ available: true }));
+    });
+  });
+
+  describe("#handleEvent.put", () => {
+    it("should dispatch `ReceivedTextChatMessage` when a text message is received", () => {
+      driver.handleEvent({
+        data: JSON.stringify({
+          data: {
+            timestamp: 1234567890123,
+            value: {
+              contentType: CHAT_CONTENT_TYPES.TEXT,
+              message: "Are you there?",
+              sentTimestamp: "2016-05-10T18:26:58.235Z"
+            }
+          },
+          path: "/chat!01234567abcdefghijkl"
+        }),
+        type: "put"
+      });
+
+      sinon.assert.called(dispatcher.dispatch);
+      sinon.assert.calledWithExactly(dispatcher.dispatch,
+        new actions.ReceivedTextChatMessage({
+          contentType: CHAT_CONTENT_TYPES.TEXT,
+          message: "Are you there?",
+          receivedTimestamp: "2009-02-13T23:31:30.123Z",
+          sentTimestamp: "2016-05-10T18:26:58.235Z"
+        }));
+    });
+  });
+
+  describe("#getServerTime", () => {
+    it("should default to now", () => {
+      expect(driver.getServerTime()).eql(0);
```

Comment:
Can you add a comment just pointing out that the default fake time for sinon is 0? Took me a while to realise what was happening since 'now' isn't normally 0 ;-)